script/tool/lib/src/license_check_command.dart

```diff
@@ -116,7 +120,15 @@ class LicenseCheckCommand extends PluginCommand {
 
   @override
   Future<void> run() async {
-    final Iterable<File> allFiles = await _getAllFiles();
+    // Create a set of absolute paths to submodule directories, with trailing
+    // separator, to do prefix matching with to test directory inclusion.
+    final Iterable<String> submodulePaths = (await _getSubmoduleDirectories())
+        .map(
+            (Directory dir) => '${dir.absolute.path}${platform.pathSeparator}');
+
+    final Iterable<File> allFiles = (await _getAllFiles()).where((File file) =>
+        !submodulePaths.any(
+            (String submodule) => file.absolute.path.startsWith(submodule)));
```

Comment:
nit: can this be replaced with 
```suggestion
        !submodulePaths.any(file.absolute.path.startsWith));
```